src/test/java/uk/gov/laa/gpfd/service/ReportTrackingTableServiceTest.java

```diff
@@ -0,0 +1,241 @@
+        // Given
+        int requestedId = 1;
+        when(mappingTableService.getDetailsForSpecificReport(requestedId)).thenReturn(null);  // Simulate empty report list
+
+        // When
+        // Then
+        assertThrows(NullPointerException.class,
+                () -> reportTrackingTableService.updateReportTrackingTable(requestedId, graphClient));
+    }
+
+    @Test
+    void shouldThrowDatabaseReadExceptionWhenDatabaseErrorOccurs() {
+        // Given
+        int requestedId = 1;
+        var reportDetails = new ReportsGet200ResponseReportListInner() {{
+            setReportName("Report 1");
+            setId(1);
+        }};
+        when(mappingTableService.getDetailsForSpecificReport(requestedId)).thenReturn(reportDetails);
+        when(userService.getUserDetails(graphClient)).thenReturn(UserDetailsTestDataFactory.aValidUserDetails());
+
+        // Simulate database error
+        doThrow(new DatabaseReadException("Database error")).when(reportTrackingTableDao).updateTrackingTable(any());
+
+        // When
+        assertThrows(DatabaseReadException.class,
+                () -> reportTrackingTableService.updateReportTrackingTable(requestedId, graphClient));
+    }
+
+    @Test
+    void shouldEnsureThreadSafetyWhenMultipleThreadsCallUpdateReportTrackingTable() throws InterruptedException {
+        // Given
+        var requestedId = 1;
+        var reportDetails = new ReportsGet200ResponseReportListInner() {{
+            setReportName("Report 1");
+            setId(1);
+        }};
+
+        when(mappingTableService.getDetailsForSpecificReport(requestedId)).thenReturn(reportDetails);
+        when(userService.getUserDetails(graphClient)).thenReturn(UserDetailsTestDataFactory.aValidUserDetails());
+
+        // When & Assert
+        Thread thread1 = new Thread(() -> reportTrackingTableService.updateReportTrackingTable(requestedId, graphClient));
+        Thread thread2 = new Thread(() -> reportTrackingTableService.updateReportTrackingTable(requestedId, graphClient));
+
+        thread1.start();
+        thread2.start();
+
+        thread1.join();
+        thread2.join();
+
+        ArgumentCaptor<ReportTrackingTable> captor = ArgumentCaptor.forClass(ReportTrackingTable.class);
+
+        // Then
+        verify(reportTrackingTableDao, times(2)).updateTrackingTable(captor.capture());
+        ReportTrackingTable capturedTrackingTable = captor.getValue();
+        assertEquals(reportDetails.getReportName(), capturedTrackingTable.reportName());
+        assertEquals("www.sharepoint.com/place-where-we-will-create-report", capturedTrackingTable.reportUrl());
+        assertEquals(reportDetails.getId(), capturedTrackingTable.mappingId());
+        assertEquals("Foo Bar", capturedTrackingTable.reportGeneratedBy());
+        assertNotNull(capturedTrackingTable.creationTime());
+    }
+
+    @Test
+    void shouldCorrectlyFormatReportUrlWhenUpdatingReportTrackingTable() {
```

Comment:
The following three tests are identical and test that reportTrackingTableDao.updateTrackingTable() is being called , instead of what the method name states is being tested.